.github/workflows/blocks-playwright.yml

```diff
@@ -58,19 +69,74 @@ jobs:
                   path: plugins/woocommerce-blocks/tests/e2e/artifacts/test-results
                   if-no-files-found: ignore
 
+            - name: Archive flaky test reports
+              uses: actions/upload-artifact@v4
+              if: ${{ !cancelled() }}
+              with:
+                  name: flaky-tests-report-shard-${{ matrix.shardIndex }}
+                  path: plugins/woocommerce-blocks/flaky-tests
+                  if-no-files-found: ignore
+
     merge-artifacts:
         # Merges all artifacts from all shards into a single zip and
         # deletes the parts. In case of a rerun, artifacts from the
-        # previous run will be retained by merging them with the new ones. 
+        # previous run will be retained by merging them with the new ones.
         name: Merge Artifacts
         if: ${{ !cancelled() }}
         needs: [blocks-playwright-tests]
         runs-on: ubuntu-latest
+        outputs:
+            has-flaky-test-report: ${{ !!steps.merge-flaky-tests-reports.outputs.artifact-id }}
         steps:
-            - name: Merge Artifacts
+            - name: Merge failures artifacts
               uses: actions/upload-artifact/merge@v4
               # Don't fail the job if there aren't any artifacts to merge.
               continue-on-error: true
               with:
                   name: failures-artifacts
+                  pattern: failures-artifacts*
                   delete-merged: true
+
+            - name: Merge flaky tests reports
+              id: merge-flaky-tests-reports
+              uses: actions/upload-artifact/merge@v4
+              continue-on-error: true
+              with:
+                  name: flaky-tests-report
+                  pattern: flaky-tests-report*
+                  delete-merged: true
+
+    create-github-issues-for-flaky-tests:
+        name: Create GitHub issues for flaky tests
+        needs: [merge-artifacts]
+        if: ${{ !cancelled() && needs.merge-artifacts.outputs.has-flaky-test-report == 'true' }}
+        runs-on: ubuntu-latest
+        permissions:
+            contents: read
+            issues: write
+        steps:
+            # We are using @wordpress/report-flaky-tests package from Gutenberg repo
+            # to create GitHub issues for flaky tests.
+            - uses: actions/checkout@v4
+              with:
+                  repository: WordPress/gutenberg
+                  ref: trunk
```

Comment:
You are correct, it's possible. Shouldn't the code of the trunk be stable? If we lock to a specific commit, we may miss bug fixes or new features that will be added to `@wordpress/report-flaky-tests`. I don't have a strong opinion, though. I can change it to specific commit.

Reply:
It's more like treating the commit SHA as the script version. Since it's tied to the [reporter script added to our codebase](https://github.com/woocommerce/woocommerce/pull/47758/files#diff-30c70629b325ae863c7c1f6c202238cc21f14b1e7f1cc528d1aee88f1be3f6f1) we might be risking some misalignment, no?

Reply:
Yeah, my reasoning was that we always want to get the latest script version as there might be some improvements to scripts or bug fixes. But I think we can go with the commit SHA too. I don't have a strong opinion. Let me make this change and use the latest trunk commit SHA from the Gutenberg repo. 🤝

Reply:
Made the changes in [dd994d9](https://github.com/woocommerce/woocommerce/pull/47758/commits/dd994d9f738e9c0566f58421f94def5dde135a40)